lib/github_changelog_generator/octo_fetcher.rb

```diff
@@ -0,0 +1,293 @@
+module GitHubChangelogGenerator
```

Comment:
I tried to maintain all the interfaces, and use similar method calls, logic flow to the original Fetcher class. A follow-up PR can be done to clean it up even more, but this hopefully aids in review.